### PR TITLE
Inventory and model compute snapshots

### PR DESCRIPTION
**Pull request by jceresini**
We expect to need to perform checks on the backup/snapshot status of existing disks. In order to do so we'll need an inventory of the snapshots from compute engine.

### CODE REVIEW

**Comment by blueandgold** [comment]
Awesome, thanks for this!  Please give me a chance to review this closely tomorrow!

File: google/cloud/forseti/services/inventory/base/gcp.py [hunk 1]
```diff
@@ -667,6 +667,19 @@ def iter_networks(self, projectid):
         for network in self.compute.get_networks(projectid):
             yield network
 
+    @create_lazy('compute', _create_compute)
+    def iter_computesnapshots(self, projectid):
```
Comment (blueandgold) [inline google/cloud/forseti/services/inventory/base/gcp.py]: To be consistent, can we just call this `iter_snapshots`?  As other methods here, do not incorporate the api naming, i.e. `compute`.

File: tests/common/gcp_api/test_data/fake_compute_responses.py [hunk 2]
```diff
@@ -1350,6 +1350,107 @@
 
 EXPECTED_NETWORK_NAME = [u"default1", u"default2"]
 
+SNAPSHOTS_LIST_PAGE1 = """
+{
+ "kind": "compute#snapshotList",
+ "id": "projects/cleardata-lab-jpc/global/snapshots",
```
Comment (blueandgold) [inline tests/common/gcp_api/test_data/fake_compute_responses.py]: Can we please anonymize this and everything else in the test data?

File: google/cloud/forseti/services/inventory/base/gcp.py (hunk 1 above)
Comment (jceresini) [reply google/cloud/forseti/services/inventory/base/gcp.py]: Happy to change it, for what its worth i used compute disks as a reference for much of what I did here, and that calls the function `iter_computedisks`.

File: tests/common/gcp_api/test_data/fake_compute_responses.py (hunk 2 above)
Comment (jceresini) [reply tests/common/gcp_api/test_data/fake_compute_responses.py]: Doh, cant believe I missed that. Thanks for catching it!

File: google/cloud/forseti/services/inventory/base/gcp.py (hunk 1 above)
Comment (blueandgold) [reply google/cloud/forseti/services/inventory/base/gcp.py]: Good point, opening an issue to us to make the naming consistent here.   :)

Comment (blueandgold) [reply google/cloud/forseti/services/inventory/base/gcp.py]: https://github.com/GoogleCloudPlatform/forseti-security/issues/1901

**Pull request merged**